---
bool success
bool grasped
bool avoid_obs
bool goal_reach